deployment/ion/ion.tf

```diff
@@ -129,6 +129,145 @@ resource "kubernetes_deployment" "ion-front-api" {
   }
 }
 
+resource "tls_private_key" "ca" {
+  algorithm = "RSA"
+  rsa_bits  = "2048"
+}
+
+resource "tls_self_signed_cert" "ca" {
+  key_algorithm   = "${tls_private_key.ca.algorithm}"
+  private_key_pem = "${tls_private_key.ca.private_key_pem}"
+
+  subject {
+    common_name  = "Ion CA"
+    organization = "Ion, Ltd"
+    country      = "GB"
+  }
+
+  validity_period_hours = 43800
+  is_ca_certificate     = true
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "client_auth",
+    "cert_signing",
+  ]
+}
+
+resource "tls_private_key" "server" {
+  algorithm = "RSA"
+  rsa_bits  = "2048"
+}
+
+resource "random_string" "server" {
+  length  = 8
+  upper   = false
+  special = false
+  number  = false
+}
+
+resource "tls_cert_request" "server" {
+  key_algorithm   = "${tls_private_key.server.algorithm}"
+  private_key_pem = "${tls_private_key.server.private_key_pem}"
+
+  subject {
+    common_name  = "ion${random_string.server.result}.${var.resource_group_location}.cloudapp.azure.com"
+    organization = "Ion, Ltd"
+    country      = "GB"
+  }
+
+  dns_names = ["ion${random_string.server.result}.${var.resource_group_location}.cloudapp.azure.com"]
+}
+
+resource "tls_locally_signed_cert" "server" {
+  cert_request_pem = "${tls_cert_request.server.cert_request_pem}"
+
+  ca_key_algorithm   = "${tls_private_key.ca.algorithm}"
+  ca_private_key_pem = "${tls_private_key.ca.private_key_pem}"
+  ca_cert_pem        = "${tls_self_signed_cert.ca.cert_pem}"
+
+  validity_period_hours = 43800
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "client_auth",
+  ]
+}
+
+resource "tls_private_key" "client" {
+  algorithm = "RSA"
+  rsa_bits  = "2048"
+}
+
+resource "tls_cert_request" "client" {
+  key_algorithm   = "${tls_private_key.client.algorithm}"
+  private_key_pem = "${tls_private_key.client.private_key_pem}"
+
+  subject {
+    common_name  = "ion${random_string.server.result}.${var.resource_group_location}.cloudapp.azure.com"
```

Comment:
Should this be taken as the output of the kubernetes service rather than reconstructed here?

Reply:
Or moved to a local as it's used in multiple places